logfire/_internal/integrations/urllib3.py

```
@@ -0,0 +1,37 @@
+import urllib3.connectionpool
+import urllib3.response
+from opentelemetry.sdk.trace import Span
+
+try:
+    from opentelemetry.instrumentation.urllib3 import RequestInfo, URLLib3Instrumentor
+except ImportError:
+    raise RuntimeError(
+        '`logfire.instrument_urllib3()` requires the `opentelemetry-instrumentation-urllib3` package.\n'
+        'You can install this with:\n'
+        "    pip install 'logfire[urllib3]'"
+    )
```

Ruff (0.15.17)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)